arch/posix/CMakeLists.txt

```diff
@@ -1,12 +1,17 @@
 zephyr_compile_options(
-  -fno-freestanding
   -m32
   -MMD
   -MP
   ${ARCH_FLAG}
   -include ${ZEPHYR_BASE}/arch/posix/include/posix_cheats.h
   )
 
+if(NOT CMAKE_C_COMPILER_ID STREQUAL "Clang") # clang errors on -fno-freestanding
```

Comment:
There are many toolchains that are not clang.
I'd suggest the inverse: If GNU, rather than if not Clang.

Reply:
@mped-oticon 

Makes sense. Fixed, have a look.